phoenix-core/src/main/antlr3/PhoenixSQL.g

```diff
@@ -704,19 +707,31 @@ column_defs returns [List<ColumnDef> ret]
     :  v = column_def {$ret.add(v);}  (COMMA v = column_def {$ret.add(v);} )*
 ;
 
+column_qualifier_counters returns [Map<String, Integer> ret]
+@init{ret = new HashMap<String,Integer>(); }
+    :   k=identifier EQ v=NUMBER {$ret.put(k, Integer.parseInt( v.getText() ));}
+        (COMMA k=identifier EQ v=NUMBER {$ret.put(k, Integer.parseInt( v.getText() ));} )*
+    ;
+
 indexes returns [List<NamedNode> ret]
 @init{ret = new ArrayList<NamedNode>(); }
     :  v = index_name {$ret.add(v);}  (COMMA v = index_name {$ret.add(v);} )*
 ;
 
 column_def returns [ColumnDef ret]
-    :   c=column_name dt=identifier (LPAREN l=NUMBER (COMMA s=NUMBER)? RPAREN)? ar=ARRAY? (lsq=LSQUARE (a=NUMBER)? RSQUARE)? (nn=NOT? n=NULL)? (DEFAULT df=expression)? (pk=PRIMARY KEY (order=ASC|order=DESC)? rr=ROW_TIMESTAMP?)?
-        { $ret = factory.columnDef(c, dt, ar != null || lsq != null, a == null ? null :  Integer.parseInt( a.getText() ), nn!=null ? Boolean.FALSE : n!=null ? Boolean.TRUE : null, 
+    :   c=column_name dt=identifier (LPAREN l=NUMBER (COMMA s=NUMBER)? RPAREN)? ar=ARRAY? (lsq=LSQUARE (a=NUMBER)? RSQUARE)? (nn=NOT? n=NULL)? (DEFAULT df=expression)? ((pk=PRIMARY KEY (order=ASC|order=DESC)? rr=ROW_TIMESTAMP?)|(COLUMN_QUALIFIER_ID cq=NUMBER))?
```

Comment:
What re the advantages of using encoded binary literals?
I believe these checks are more readable this way:
`cq.getValue() < QueryConstants.ENCODED_CQ_COUNTER_INITIAL_VALUE`

Reply:
We can use that internal representation independent of the user-facing format.

I've just verified, that the "COLUMN_QUALIFIER" syscat column always stores the qualifier, whether we use encoded qualifiers, or unencoded ones.

On the other hand, there is no way to specify the unencoded COLUMN_QUALIFER directly, so that does not rule out using the integer values here.

We should make sure to use a name that makes it clear this is only for encoded qualifiers, like 
ENCODED_QUALIFER.